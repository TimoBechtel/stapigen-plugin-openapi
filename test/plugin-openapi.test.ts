jest.mock('fs');

import { fs, vol } from 'memfs';
import { generateApi } from 'stapigen';
import type { Config } from 'stapigen/dist/config';
import pluginOpenAPI from '../src/';

const testDir = 'test';

beforeEach(() => {
	vol.mkdirSync(testDir, { recursive: true });
	vol.fromJSON(
		{
			'2020/file.md': 'file without a month + day',
			'2020/04/10/lorem.md': '',
			'2020/04/13/randomNotes.md': '',
			'2020/04/13/thoughts.md': '',
		},
		'test/example/input'
	);
});

afterEach(() => {
	vol.reset();
});

test('creates an openapi specification from stapigen collections', async () => {
	const config: Config = {
		input: {
			dir: 'test/example/input',
			schema: ':year/:month/:day',
		},
		output: {
			dir: testDir,
			schema: ':month/:day/',
		},
		parser: [
			{
				extensions: ['.md'],
				parse: ({ content }) => ({ text: content }),
			},
		],
		plugins: [
			pluginOpenAPI({
				dir: testDir,
				spec: {
					info: {
						title: 'Test',
						version: '1.0.0',
					},
					openapi: '3.0.0',
				},
			}),
		],
	};

	await generateApi(config);

	expect(fs.existsSync(`${testDir}/openapi.json`)).toBe(true);
	expect(fs.readFileSync(`${testDir}/openapi.json`, 'utf8')).toMatchSnapshot();
});
