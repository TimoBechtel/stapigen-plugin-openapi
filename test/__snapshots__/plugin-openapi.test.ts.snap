// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates an openapi specification from stapigen collections 1`] = `
"{
  \\"info\\": {
    \\"title\\": \\"Test\\",
    \\"version\\": \\"1.0.0\\"
  },
  \\"openapi\\": \\"3.0.0\\",
  \\"paths\\": {
    \\"/04/10/index.json\\": {
      \\"summary\\": \\"Get all items\\",
      \\"get\\": {
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"Success\\"
          }
        }
      }
    },
    \\"/04/10/{id}.json\\": {
      \\"get\\": {
        \\"summary\\": \\"Get a single item\\",
        \\"parameters\\": [
          {
            \\"name\\": \\"id\\",
            \\"in\\": \\"path\\",
            \\"required\\": true,
            \\"schema\\": {
              \\"type\\": \\"string\\"
            }
          }
        ],
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"Success\\"
          }
        }
      }
    },
    \\"/04/13/index.json\\": {
      \\"summary\\": \\"Get all items\\",
      \\"get\\": {
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"Success\\"
          }
        }
      }
    },
    \\"/04/13/{id}.json\\": {
      \\"get\\": {
        \\"summary\\": \\"Get a single item\\",
        \\"parameters\\": [
          {
            \\"name\\": \\"id\\",
            \\"in\\": \\"path\\",
            \\"required\\": true,
            \\"schema\\": {
              \\"type\\": \\"string\\"
            }
          }
        ],
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"Success\\"
          }
        }
      }
    },
    \\"/_/_/index.json\\": {
      \\"summary\\": \\"Get all items\\",
      \\"get\\": {
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"Success\\"
          }
        }
      }
    },
    \\"/_/_/{id}.json\\": {
      \\"get\\": {
        \\"summary\\": \\"Get a single item\\",
        \\"parameters\\": [
          {
            \\"name\\": \\"id\\",
            \\"in\\": \\"path\\",
            \\"required\\": true,
            \\"schema\\": {
              \\"type\\": \\"string\\"
            }
          }
        ],
        \\"responses\\": {
          \\"200\\": {
            \\"description\\": \\"Success\\"
          }
        }
      }
    }
  }
}"
`;
